src/components/shared/VirtualList.js

```diff
@@ -244,22 +250,14 @@ type Geometry = {
   innerRectY: CssPixels,
 };
 
-class VirtualList extends React.PureComponent<VirtualListProps> {
-  _container: ?HTMLDivElement;
-  _inner: ?VirtualListInner;
+class VirtualList<Item> extends React.PureComponent<VirtualListProps<Item>> {
+  _container: {| current: HTMLDivElement | null |} = React.createRef();
```

Comment:
I kind of want a named type for the new React refs. Something like:

```
type ReactRef<T> => {|current: null | T|};
```

Reply:
plot twist, this does exist:
https://github.com/facebook/flow/blob/0938da8d7293d0077fbe95c3a3e0eebadb57b012/lib/react.js#L190-L196

But:
* the fact that it supports the new style isn't documented
* it would be painful to use with the union

Should we roll our own? 
I think that most of the time we could probably rely on `createRef` returning the right thing, but that in this case this is buggy because of the parametric type.

Reply:
>  I think that most of the time we could probably rely on createRef returning the right thing, but that in this case this is buggy because of the parametric type.

That's fine if it won't come up much. I don't think we want the built-in one, since it seems to not be quite correct for our case, as the union is overly broad.